src/WebJobs.Script/Diagnostics/FileTraceWriter.cs

```diff
@@ -39,90 +40,141 @@ public FileTraceWriter(string logFilePath, TraceLevel level, LogType logType) :
             _instanceId = GetInstanceId();
             _logType = logType;
             _logDirectory = new DirectoryInfo(logFilePath);
-
-            // start a timer to flush accumulated logs in batches
-            _flushTimer = new Timer
-            {
-                AutoReset = true,
-                Interval = LogFlushIntervalMs
-            };
-            _flushTimer.Elapsed += OnFlushLogs;
-            _flushTimer.Start();
         }
```

Comment:
The timer is no longer started at construction. We delay the timer initialization until there's something to log. If no logging is ever done, the timer will never be setup.